### PR TITLE
rpc: Remove deprecated bumpfee behavior

### PR DESCRIPTION
**Pull request by achow101**
Removes the deprecation message, behavior, and test.

This was marked for removal in 22.0.

### CODE REVIEW

File: test/functional/rpc_deprecated.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 """Test deprecation of RPC calls."""
 from test_framework.test_framework import BitcoinTestFramework
-from test_framework.util import assert_raises_rpc_error, find_vout_for_address
+ifrom test_framework.util import assert_raises_rpc_error
```
Comment (jonatack) [inline test/functional/rpc_deprecated.py:7]:
```
    ifrom test_framework.util import assert_raises_rpc_error
          ^
SyntaxError: invalid syntax
```

Comment (achow101) [reply test/functional/rpc_deprecated.py:7]: oops. fixed

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -3449,10 +3449,7 @@ static RPCHelpMan bumpfee_helper(std::string method_name)
     CWallet* const pwallet = wallet.get();
 
     if (pwallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS) && !want_psbt) {
-        if (!pwallet->chain().rpcEnableDeprecated("bumpfee")) {
-            throw JSONRPCError(RPC_METHOD_DEPRECATED, "Using bumpfee with wallets that have private keys disabled is deprecated. Use psbtbumpfee instead or restart bitcoind with -deprecatedrpc=bumpfee. This functionality will be removed in 0.22");
-        }
-        want_psbt = true;
+        throw JSONRPCError(RPC_METHOD_DEPRECATED, "bumpfee is not available with wallets that have private keys disabled. Use psbtbumpfee instead.");
```
Comment (jonatack) [inline src/wallet/rpcwallet.cpp:3452]: Might be good to move the assert_raises_rpc_error test into test/functional/wallet_bumpfee.py.

**Review by jonatack — COMMENTED** [review]
utACK, modulo test suggestion

File: src/wallet/rpcwallet.cpp (hunk 2 above)
Comment (achow101) [reply src/wallet/rpcwallet.cpp]: Done

**Comment by hafixo** [comment]
Tarik penggabungan diminta #20890

**Review by promag — COMMENTED** [review]
ACK ea0a7ec949f0f7e212f0d8819f7a54cad2258bdd, maybe add need release notes tag.

**Review by hendeyk — APPROVED** [review]

**Pull request merged**